apps/ensadmin/src/components/connections/connection-selector.tsx

```diff
@@ -93,8 +95,11 @@ export function ConnectionSelector() {
   };
 
   const handleRemove = (url: string) => {
-    if (url === preferredEnsNodeUrl()) return;
-    removeConnection.mutate({ url });
+    const connectionToBeRemoved = connections.find((c) => c.isDefault === false && c.url === url);
+
+    if (connectionToBeRemoved) {
+      removeConnection.mutate(connectionToBeRemoved);
+    }
```

Comment:
Only remove connection if it can be removed (found by URL and not marked as a default one).